libs/logging/include/hpx/logging/format/named_write.hpp

```diff
@@ -399,8 +399,10 @@ This will just configure "file" twice, ending up with writing only to "two.txt"
             std::stringstream out;
             m_format(out, msg);
 
+#if !defined(HPX_COMPUTE_HOST_CODE)
             message formatted(std::move(out));
             m_destination(formatted);
+#endif
```

Comment:
I'd expect there would be an `#else` here in some form. This effectively disables formatting in logging.

Reply:
Well, yes and no. Unless we provide a way to redirect messages to `printf` in device code the logging capabilities can't be used in device code anyway. This just makes it compile but does not work exactly as you say. Hence my earlier suggestion to just not build logging at all when compiling device code.

Reply:
FWIW, I wouldn't expect for logging to be invoked in device code. So I think the current solution is good enough. We could add a (non-HPX)  `assert()` however to make sure.

Reply:
Indeed. `HPX_ASSERT` uses an `assert` which works in device code btw, so it's safe to use `HPX_ASSERT` directly (there's currently no error message, though).

Reply:
> HPX_ASSERT uses an assert

Hmm, does it? I thought we used our own (https://github.com/STEllAR-GROUP/hpx/blob/master/libs/assertion/include/hpx/assertion.hpp#L59-L65).

Reply:
Note the trailing underscore on the definition you linked ;) `HPX_ASSERT` itself is defined differently if compiling device code: https://github.com/STEllAR-GROUP/hpx/blob/aa59c1bc6e97410a5e7db009feec5693fe0cf4dc/libs/assertion/include/hpx/assertion.hpp#L68-L71.

Reply:
Yes, you're certainly right!